src/alexandria3k/data_sources/issn_subject_codes.py

```diff
@@ -0,0 +1,221 @@
+        name.
+    :type attach_databases: list, optional
+
+    """
+
+    def __init__(
+        self,
+        data_source=None,
+        sample=lambda n: True,
+        attach_databases=None,
+        config_path=None,
+    ):
+        # Check if the environment variable is set or not
+        if "PYBLIOMETRICS_CONFIG_FILE" not in os.environ:
+
+            def get_config_path(config_path):
+                if config_path:
+                    return config_path
+                config_path = os.path.expanduser("~/.config/pybliometrics.cfg")
+                if not os.path.exists(config_path):
+                    raise FileNotFoundError(
+                        f"Configuration file not found at {config_path}"
+                    )
+                debug.log("config-file", f"Using config file at {config_path}")
+                return config_path
+
+            # Set the configuration file path
+            config_path = get_config_path(config_path)
+            os.environ["PYBLIOMETRICS_CONFIG_FILE"] = config_path
+
+        self.sample = sample
+        super().__init__(
+            VTSource(issn_subject_codes_table, data_source, sample),
+            [issn_subject_codes_table],
+            attach_databases,
+        )
+
+    def execute_sql_query(self, cursor, script):
+        """Execute the specified SQL query and return the results."""
+        cursor.execute(script)
+        issns = [row[0] for row in cursor.fetchall()]
+        return issns
+
+    def debug_progress_bar(self, current_progress, total_length):
+        """Print a progress bar"""
+        percent = current_progress / total_length * 100
+        progress_marker = int(
+            PROGRESS_BAR_LENGTH * current_progress / total_length
+        )
+        progress_bar = "#" * progress_marker + "-" * (
+            PROGRESS_BAR_LENGTH - progress_marker
+        )
+        debug.log(
+            "progress_bar",
+            f"\r[{progress_bar}] {percent:.2f}% | "
+            f"Downloaded {current_progress} out of {total_length} ISSNs",
+            end="",
+        )
+
+    def fetch_subject_codes(self, writer, issns):
+        """Fetch the subject codes for the specified ISSNs."""
+        total_issns = len(issns)
+        for index, issn in enumerate(issns):
+            self.debug_progress_bar(index + 1, total_issns)
+            query = {"issn": issn}
+            try:
+                serial_search = SerialSearch(query=query, view="STANDARD")
+                results = list(serial_search.results)
+
+                subject_area_codes_set = set()
+
+                for result in results:
+                    if (
+                        "subject_area_codes" in result
+                        and result["subject_area_codes"]
+                    ):
+                        subject_area_codes = result[
+                            "subject_area_codes"
+                        ].split(";")
+                        subject_area_codes_set.update(subject_area_codes)
```

Comment:
Do you have to store everything in memory and then write it? Can't you stream it all directly to the file?